src/features/machines/backgroundQuoterMachine.ts

```diff
@@ -85,27 +92,64 @@ async function pollQuoteLoop(
   delayMs: number,
   onResult: (result: AggregatedQuote) => void
 ): Promise<void> {
+  let lastPropagatedResultRequestedAt: number | null = null
+
   while (!signal.aborted) {
-    const quote = await queryQuote({
-      tokensIn: isBaseToken(quoteInput.tokenIn)
-        ? [quoteInput.tokenIn.defuseAssetId]
-        : quoteInput.tokenIn.groupedTokens.map((token) => token.defuseAssetId),
-      tokensOut: isBaseToken(quoteInput.tokenOut)
-        ? [quoteInput.tokenOut.defuseAssetId]
-        : quoteInput.tokenOut.groupedTokens.map((token) => token.defuseAssetId),
-      amountIn: quoteInput.amountIn,
-      balances: quoteInput.balances,
-    }).catch((e) => {
-      console.error("Error querying quote", e)
-      return null
-    })
+    const requestedAt = Date.now()
 
-    if (quote == null || signal.aborted) {
-      return
-    }
+    queryQuote(
+      {
+        tokensIn: getUnderlyingDefuseAssetIds(quoteInput.tokenIn),
```

Comment:
Looks like it’s more accurate to say this property is `tokenInIds` rather than a `tokensIn` object.

Reply:
Yeah, I also thought about it, but I didn't like separation of words "token" and "id", so my second thought was to have `tokenIdsIn` but it looked even worse, so I decided to go with just `tokensIn`